src/main/java/com/bunq/sdk/security/SecurityUtils.java

```diff
@@ -412,16 +383,14 @@ private static String generateMethodAndEndpointString(BunqRequestBuilder request
   }
 
   private static String generateRequestHeadersSortedString(BunqRequestBuilder bunqRequestBuilder) {
-    return Arrays.stream(bunqRequestBuilder.getAllHeaderAsArray())
-        .filter(
+    return BunqBasicHeader.collectForSigning(bunqRequestBuilder.getAllHeader()
```

Comment:
I was in the process of refactoring this in #42. As you are touching it, please refactor to make it java 7 compatible.